### ISSUE TITLE
Add Host Blocker Similar to AdAway

### ISSUE THREAD

**Issue by cwmke** [copperhead/bugtracker#343]
I am suggesting adding a host blocker with similar functionality to [AdAway](https://github.com/AdAway/AdAway). It essentially downloads a lists of hosts to block and then blocks them system-wide. The user can add or remove lists as they please.  There is also an user modifiable whitelist and blacklist available.

I imagine this is something many users would love to have built-in but it could be disabled by default in order to create issues for users who would find it unfriendly.

I noticed in #123 @thestinger stated,
```
CopperheadOS used to have domain and IP blacklists but it's too heavy handed and user unfriendly.
```
#184 is similar but was looking at a very specific use case.

**Comment by thestinger** [comment]
There's #7 which was filed a long time ago. Maybe someone is interested in implementing it. It's an extremely low priority right now.

**Comment by aes512** [comment]
That sucks. /etc/hosts file based ad blocking goes a long f**king way on Android. It seems like everything important in this project is low prio. I guess I'll stick to building my own ROMs, I really had hope that this would become my daily driver :(

**Comment by thestinger** [comment]
CopperheadOS has a single developer so adding features can't be a high priority right now. If people want features to be implemented, they're free to step up and contribute. Right now, no one contributes to the project. Domain blocking via /system/etc/hosts isn't something that can be done though. It has to be tackled in a more sophisticated way where it's not mandatory and the list of blocked hosts can be modified.

**Comment by thestinger** [comment]
It's not like there are any apps on F-Droid with ads anyway... and blocking content in browsers works better via browser extensions or direct integration, since there's a lot more to it than simply blocking domains. Blocking domains from resolving also has terrible UX. It silently kills connections to the domains... it isn't really suited to what CopperheadOS wants to be. uBlock Origin, etc. are a lot better suited to it, so Chromium extension support is what's really needed.

**Comment by aes512** [comment]
We will take the (not so) "terrible" UX over some artificial idea of some perfect ROM ecosphere, because it actually produces a positive result -- being that the MAJORITY of "harmless" and malicious ads (and other unwanted web-based connections) get blocked. It is a tried and true mechanism that spans a lot further than the data mining google derived browser that is baked into the ROM, but seeing the benefit would require having the ability to see the 100K' view of things, which clearly isn't this project's scope.

A "hardened" mobile OS is a bit of a joke without additional support for mechanisms that protect it's users. Your ROM just becomes a little more difficult for criminals to pop, but it lends itself to being a far broader target for data mining/profiling and a vehicle for surveillance. It's like putting a glass windows on a steel house, but whatever - continue missing the point, you have your priorities and I'm not going to try and change them because I for one do not care.

**Comment by thestinger** [comment]
Chromium is the WebView, so it's going to be there whether or not it's our chosen browser. It's not only the most secure browser available but also shares 99.99% of the code with the WebView, Soon, both will even be provided by the same apk. It really makes no sense for us to include any other browser since it would add significant attack surface and none has a comparable sandbox.

Chromium doesn't do profiling / data mining especially since the Google features mostly don't work on CopperheadOS due to the lack of Google Play. You can use another search engine if that's what you're worried about. It's ridiculous to get stuck on the fact that it's Google's code. You're talking about an Android-based operating system here. It's all Google's code. Leaving out Chromium would only mean leaving out the tiny portion of Chromium's code that provides the Android browser UX. It would still be present as the WebView.

Lack of extension support in Chromium is something that can and should be rectified. Content filtering at a browser level is far better than doing it at the DNS level. It can block far more and it's usable which is key for our goals. As soon as Chromium supports extensions, we'll happily file an issue about implementing a way to have uBlock Origin installed by default in new profiles.

**Comment by thestinger** [comment]
Since I'm not particularly interested in the opinions of entitled morons that are too lazy to contribute anything of value to the world, you've earned the prestigious award of being the first person that I've ever blocked on GitHub (from the Copperhead organization).
